feat: share Connection & DNS Cache to `CURLRequest` (#9557)
codeigniter4/CodeIgniter4: user_guide_src/source/libraries/curlrequest.rst

Config for CURLRequest
**********************

.. _curlrequest-sharing-connection:

Sharing Connection
==================

.. versionadded:: 4.7.0

> If you want to share connection between requests, set ``$shareConnection`` with array constant `CURL_LOCK_DATA_* <https://www.php.net/manual/en/curl.constants.php#constant.curl-lock-data-connect>`_ in **app/Config/CURLRequest.php**:

Reviewer: It would be good to mention that these are the default values.

Config for CURLRequest
**********************

.. _curlrequest-sharing-connection:

Sharing Connection
==================

.. versionadded:: 4.7.0

> .. important:: By default this behavior already enabled with two constants ``CURL_LOCK_DATA_CONNECT`` and ``CURL_LOCK_DATA_DNS``.

Reviewer: ```suggestion
By default, this option is enabled with the constants ``CURL_LOCK_DATA_CONNECT`` and ``CURL_LOCK_DATA_DNS``.
```

Not sure if we need a special box for this.
